singer_sdk/connectors/sql.py

```diff
@@ -441,21 +468,34 @@ def discover_catalog_entry(
 
         # Initialize columns list
         table_schema = th.PropertiesList()
-        for column_def in inspected.get_columns(table_name, schema=schema_name):
-            column_name = column_def["name"]
-            is_nullable = column_def.get("nullable", False)
-            jsonschema_type: dict = self.to_jsonschema_type(
-                t.cast(sqlalchemy.types.TypeEngine, column_def["type"]),
-            )
-            table_schema.append(
-                th.Property(
-                    name=column_name,
-                    wrapped=th.CustomType(jsonschema_type),
-                    required=not is_nullable,
-                ),
-            )
+        with warnings.catch_warnings(record=True) as inspection_warnings:
+            for column_def in inspected.get_columns(table_name, schema=schema_name):
+                column_name = column_def["name"]
+                is_nullable = column_def.get("nullable", False)
+                jsonschema_type: dict = self.to_jsonschema_type(
+                    t.cast(sqlalchemy.types.TypeEngine, column_def["type"]),
+                )
+                table_schema.append(
+                    th.Property(
+                        name=column_name,
+                        wrapped=th.CustomType(jsonschema_type),
+                        required=not is_nullable,
+                    ),
+                )
+        if len(inspection_warnings) > 0:
+            for line in inspection_warnings:
+                expanded_msg: str = (
```

Comment:
For this bits, wdyt about using the builtin [`logging.captureWarnings`](https://docs.python.org/3/library/logging.html#logging.captureWarnings)?

We'd probably want to call it whenever a plugin's CLI is invoked, so in the body of https://github.com/meltano/sdk/blob/046044f9de808eff215681a3c8fd0009efe9fbfb/singer_sdk/plugin_base.py#L513-L531

Reply:
Done a while ago in https://github.com/meltano/sdk/commit/2a64cf612af63cec99efacf5aa3942bc685fdb46